harness/determined/constants.py

```diff
@@ -24,16 +26,39 @@
     "optimizations": DEFAULT_OPTIMIZATIONS,
 }
 
+# Allow users to override the ports using the environment variables, in case
+# there are other processes running on the compute node which are already using
+# our default port. Users will likely set these in their "startup-hook.sh".
+# Below is an example how to set these in the experiment's YAML file for testing.
+#
+# environment:
+#   environment_variables:
+#      - DTRAIN_SSH_PORT=13000
+#      - HOROVOD_GLOO_RENDEZVOUS_PORT=13010
+#      - INTER_TRAIN_PROCESS_COMM_PORT_1=13020
+DTRAIN_SSH_PORT_ENV_VAR = "DTRAIN_SSH_PORT"
+HOROVOD_GLOO_RENDEZVOUS_PORT_ENV_VAR = "HOROVOD_GLOO_RENDEZVOUS_PORT"
+INTER_TRAIN_PROCESS_COMM_PORT_1_ENV_VAR = "INTER_TRAIN_PROCESS_COMM_PORT_1"
+
 # TODO (DET-1189): Use port registry to allocate ssh port.
 # SSH port used for agents during dtrain (currently used with horovod and deepspeed backend).
-DTRAIN_SSH_PORT = 12350
+# Note: Use of str() is to prevent lint complaint:
+#   error: Argument 1 to "int" has incompatible type "Optional[str]"
```

Comment:
I'd prefer you leave these `str()` comments out.  This doesn't quite raise enough of an eyebrow to justify a comment, and certainly not enough to justify three comments in close proximity.

Reply:
Will change.